Lazariev_lab_3.py

```diff
@@ -0,0 +1,120 @@
+import math
+import random
+import matplotlib.pyplot as plt
+
+print("Lab 3 by Viktor Lazariev, IKM-220d")
+# 1) За наведеним вище алгоритмом промоделювати процес навчання та роботи двошарового персептрону.
+# 2) Дослідити вплив початкових умов вагових коефіцієнтів на ефективність навчання:
+# провести моделювання для рекомендованих значень та для інших, вільно обраних значень.
+# Порівнювати варіанти за кількістю епох навчання, необхідних для нормальної роботи мережі.
+# 3) Дослідити вплив коефіцієнту швидкості навчання на його ефективність.
+# 4) Дослідити залежність ефективності алгоритму навчання від параметра k, що входить до ступеню в функції активації.
+# 5) Для рекомендованого варіанту побудувати залежність E* від кількості епох навчання та визначити оптимальну кількість.
+# У звіті про лабораторну роботу відобразити результати за пп.2-5 та навести найкращі вагові коефіцієнти.
+# Зробити відповідні висновки.
+
+# X_input=input("enter input vector X: ")
+# print("X_input= ", X_input)     #00,01,10,11
+
+def Fu(u):# F(u)
```

Comment:
An incomprehensible function, it is better to change it.
```suggestion
def sigmoid_activation(u)
```